apps/laboratory/src/pages/api/auth/[...nextauth].ts

```diff
@@ -0,0 +1,111 @@
+import type { SIWESession } from '@web3modal/core'
+import type { NextApiRequest, NextApiResponse } from 'next'
+import type { DefaultSession } from 'next-auth'
+import nextAuth from 'next-auth'
+import credentialsProvider from 'next-auth/providers/credentials'
+import { getCsrfToken } from 'next-auth/react'
+import { SiweMessage } from 'siwe'
+import { ethers } from 'ethers'
+
+declare module 'next-auth' {
+  interface Session extends SIWESession {
+    user?: DefaultSession['user']
+  }
+}
+
+const nextAuthSecret = process.env['NEXTAUTH_SECRET']
+const nextAuthUrl = process.env.NEXTAUTH_URL
+
+/*
+ * For more information on each option (and a full list of options) go to
+ * https://next-auth.js.org/configuration/options
+ */
+export default async function auth(req: NextApiRequest, res: NextApiResponse) {
+  if (!nextAuthUrl) {
```

Comment:
If I move them up then I have to cast `nextAuthUrl` and `nextAuthSecret` to as string because they're not in scope and I also have to add `// eslint-disable-next-line @typescript-eslint/no-non-null-assertion` which is not cleaner than this. 

I'll move the variables within the handler if you prefer.